lib/linter.js

```diff
@@ -280,30 +280,6 @@ function getDirectiveComments(filename, ast, ruleMapper) {
     };
 }
 
-/**
- * Normalize ECMAScript version from the initial config
- * @param  {number} ecmaVersion ECMAScript version from the initial config
- * @param  {boolean} isModule Whether the source type is module or not
- * @returns {number} normalized ECMAScript version
- */
-function normalizeEcmaVersion(ecmaVersion, isModule) {
-
-    // Need at least ES6 for modules
-    if (isModule && (!ecmaVersion || ecmaVersion < 6)) {
-        return 6;
-    }
-
-    /*
-     * Calculate ECMAScript edition number from official year version starting with
-     * ES2015, which corresponds with ES6 (or a difference of 2009).
-     */
-    if (ecmaVersion >= 2015) {
-        return ecmaVersion - 2009;
```

Comment:
Removal of this normalization may be an unintentional breaking change for custom parsers and scope managers. At least, `eslint-scope` doesn't support the year form.

Reply:
good catch! 

it may be a breaking, will add it back. Though I'm sure we should normalize a 3rd-party parser option.